File: python/leetcode/136-single-number.py
class Solution:
    def singleNumber(self, nums: List[int]) -> int:
        hash = {}
        for i in range(len(nums)):
            if nums[i] not in hash:
                hash[nums[i]] = 1
            else:
                hash[nums[i]] += 1

        for key, val in hash.items():
            if val == 1:
                return key

File: c#/leetcode/136-single-number.cs
public class Solution
{
  public int SingleNumber(int[] nums)
  {
    var dict = new Dictionary<int, int>();

    foreach (int num in nums)
    {
      if (!dict.ContainsKey(num))
      {
        dict.Add(num, 1);
      }
      else
      {
        dict[num]++;
      }
    }

    int[] res = dict.Where(x => x.Value == 1).Select(x => x.Key).ToArray();
    return res[0];
  }
}